tools/functional-tester/etcd-tester/main.go

```diff
@@ -81,29 +85,54 @@ func main() {
 	if err := c.bootstrap(); err != nil {
 		plog.Fatal(err)
 	}
-	defer c.Terminate()
-
-	failures := []failure{
-		newFailureKillAll(),
-		newFailureKillMajority(),
-		newFailureKillOne(),
-		newFailureKillLeader(),
-		newFailureKillOneForLongTime(),
-		newFailureKillLeaderForLongTime(),
-		newFailureIsolate(),
-		newFailureIsolateAll(),
-		newFailureSlowNetworkOneMember(),
-		newFailureSlowNetworkLeader(),
-		newFailureSlowNetworkAll(),
+
+	if *terminate {
+		defer c.Terminate()
 	}
 
 	// ensure cluster is fully booted to know failpoints are available
 	c.WaitHealth()
-	fpFailures, fperr := failpointFailures(c)
-	if len(fpFailures) == 0 {
-		plog.Infof("no failpoints found (%v)", fperr)
+
+	failures := make([]failure, 0)
+
+	if failureTypes != nil && *failureTypes != "" {
+		fails := strings.Split(*failureTypes, ",")
```

Comment:
split this into a separate function `func makeFailures(types string, c *cluster)`?